### PR TITLE
Improve boundary preview

### PR DESCRIPTION
**Pull request by H-Yulong**
Now you can:
- Click on the "Source/Target" text to switch between source/target
- Click on the diagram to recover it (if the workspace is empty)
- Remove the source/target

### CODE REVIEW

**Comment by calintat** [comment]
Thanks Yulong, this looks nice!

**Comment by calintat** [comment]
This looks great, thanks Yulong! I think we can merge this now and will hopefully fix #609 soon.

**Comment by H-Yulong** [comment]
Great! I think it might be better to close #586 after #609 is fixed.

File: homotopy-web/src/app/boundary.rs [hunk 1]
```diff
@@ -78,6 +100,7 @@ impl BoundaryPreview {
                     diagram={ctx.props().boundary.diagram.clone()}
                     id="boundary__preview"
                     signature={ctx.props().signature.clone()}
+                    dimensions={Some((160.0, 160.0))}
```
Comment (calintat) [inline homotopy-web/src/app/boundary.rs:103]: Can you please rename this to `max_dimensions`? Or maybe split it into `max_width` and `max_height`?

Comment (H-Yulong) [reply homotopy-web/src/app/boundary.rs]: Sure, just committed.

Comment (calintat) [reply homotopy-web/src/app/boundary.rs]: Thanks!

**Pull request merged**